core/src/main/java/org/apache/brooklyn/core/location/cloud/CloudLocationConfig.java

```diff
@@ -79,7 +81,13 @@
             + "if 'false', will default to the node's first public IP (or privae if no public IPs); "
             + "if 'true' uses default duration; otherwise accepts a time string e.g. '5m' (the default) or a number of milliseconds", "5m");
 
-    public static final ConfigKey<String> WAIT_FOR_SSHABLE = ConfigKeys.newStringConfigKey("waitForSshable", 
+    ConfigKey<Predicate<HostAndPort>> POLL_FOR_FIRST_REACHABLE_ADDRESS_PREDICATE = ConfigKeys.newConfigKey(new TypeToken<Predicate<HostAndPort>>(){}, "pollForFirstReachableAddress.predicate",
+            "Predicate<HostAndPort> implementation which checks whether machine is up or not.");
+
+    ConfigKey<Class<Predicate<HostAndPort>>> POLL_FOR_FIRST_REACHABLE_ADDRESS_PREDICATE_TYPE = ConfigKeys.newConfigKey(new TypeToken<Class<Predicate<HostAndPort>>>(){}, "pollForFirstReachableAddress.predicate.type",
```

Comment:
See [here](https://github.com/apache/brooklyn-library/blob/master/software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/cassandra/CassandraDatacenter.java#L77) for an working example.